Newsletters: add 289 (2024-02-14)

harding opened the pull request:
- [ ] Lede, releases/RCs, topic entries @harding 
- [x] Bitcoin Core PR Review Club @LarryRuane

bitschmidty commented on _posts/en/newsletters/2024-02-14-newsletter.md:
```diff
@@ -0,0 +1,200 @@
+{% include linkers/issues.md v=2 issues="28948,6785,2818,2816,8338,2856,2442,2451,1131,1096" %}
+[fsfe]: https://fsfe.org/activities/ilovefs/index.en.html
+[fsf]: https://www.fsf.org/blogs/community/i-love-free-software-day-is-here-share-your-love-software-and-a-video
+[sanders future]: https://delvingbitcoin.org/t/v3-and-some-possible-futures/523/2
```
```suggestion
[sanders future]: https://delvingbitcoin.org/t/v3-and-some-possible-futures/523
```

kanzure commented:
https://lists.linuxfoundation.org/pipermail/bitcoin-dev/2024-February/022327.html

bitschmidty commented on _posts/en/newsletters/2024-02-14-newsletter.md:
```diff
@@ -0,0 +1,200 @@
+  anchor outputs] LN commitment and fee-bumping transactions (see
+  [Newsletter #286][news286 imbued] for the underlying _imbued v3_
+  proposal).  In short, he recorded 14,124 transactions in 2023 that
+  looked like anchor spends.  Of those,
```
Maybe

```suggestion
  looked like anchor spends.  Of those:
```

bitschmidty commented on _posts/en/newsletters/2024-02-14-newsletter.md:
```diff
@@ -0,0 +1,200 @@
+    under the v3 rules.
+
+  - About 2.1% <!-- 302/14124 --> had more than one parent (e.g.,
+    attempts to batch CPFP spends).  LN wallets do this for efficiency
```
```suggestion
    attempts to batch [CPFP][topic cpfp] spends).  LN wallets do this for efficiency
```

bitschmidty commented on _posts/en/newsletters/2024-02-14-newsletter.md:
```diff
@@ -0,0 +1,200 @@
+    This is also behavior that LN wallets would need to change.
+    Daftuar's further research showed that nearly all anchor spends were
+    less than 500 vbytes, potentially suggesting that the v3 size limit
+    could be reduced.  This would make it less expensive for a defender
```
Relevant?
https://bitcoinops.org/en/newsletters/2024/01/03/#v3-transaction-pinning-costs

harding replied:
Yes, but IIRC a couple reviewers on that newsletter found my description a bit hard to follow, so I think I'm going to stick with just the link to the pinning topic page.  Thanks for suggesting and finding that link, though!

bitschmidty commented on _posts/en/newsletters/2024-02-14-newsletter.md:
```diff
@@ -0,0 +1,200 @@
+repo]._
+
+- [Bitcoin Core #28948][] adds support for [version 3 transaction
+  relay][topic v3 transaction relay] allowing any v3 transaction that
```
```suggestion
  relay][topic v3 transaction relay], allowing any v3 transaction that
```

bitschmidty commented on _posts/en/newsletters/2024-02-14-newsletter.md:
```diff
@@ -0,0 +1,200 @@
+  Core's wallet will not currently automatically allow the UTXO to be
+  spent.  Eclair can independently detect a double spend of the parent
+  transaction and it will now tell Bitcoin Core's wallet to manually
+  unlock the UTXO so it can be spent again.
```
```suggestion
  abandon the transaction so the UTXO can be spent again.
```

And perhaps link to https://developer.bitcoin.org/reference/rpc/abandontransaction.html

LarryRuane commented on _posts/en/newsletters/2024-02-14-newsletter.md:
```diff
@@ -0,0 +1,283 @@
+
+- **Ideas for relay enhancements after cluster mempool is deployed:**
+  Gregory Sanders [posted][sanders future] to Delving Bitcoin several
+  ideas for allowing individual transactions to opt-in certain mempool
```
```suggestion
  ideas for allowing individual transactions to opt-in to certain mempool
```

LarryRuane commented on _posts/en/newsletters/2024-02-14-newsletter.md:
```diff
@@ -0,0 +1,283 @@
+
+  - About 2.1% <!-- 302/14124 --> had more than one parent (e.g.,
+    attempts to batch CPFP spends).  LN wallets do this for efficiency
+    when closing more than channel within a short amount of time.  They
```
```suggestion
    when closing more than one channel within a short amount of time.  They
```

LarryRuane commented on _posts/en/newsletters/2024-02-14-newsletter.md:
```diff
@@ -0,0 +1,283 @@
+    widespread attacks on the network yet, so it’s hard to figure out
+    what value would be 'better'."
+
+  Although additional discussion and research on this topic is expected,
```
```suggestion
  Although additional discussion and research on this topic are expected,
```

LarryRuane commented on _posts/en/newsletters/2024-02-14-newsletter.md:
```diff
@@ -0,0 +1,283 @@
+  believes it can safely spend by detecting some cases when an existing
+  unconfirmed transaction is very unlikely to become confirmed.  Eclair
+  uses Bitcoin Core's wallet to manage its UTXOs for onchain spending,
+  including for fee bumping transactions.  When a UTXO controlled by the
```
```suggestion
  including for fee-bumping transactions.  When a UTXO controlled by the
```

LarryRuane commented on _posts/en/newsletters/2024-02-14-newsletter.md:
```diff
@@ -0,0 +1,283 @@
+
+- [Eclair #2816][] allows the node operator to choose the maximum amount
+  they're willing to spend on an [anchor output][topic anchor outputs]
+  in order to get a commitment transaction confirmed.  Previously Eclair
```
```suggestion
  to get a commitment transaction confirmed.  Previously Eclair
```

LarryRuane commented on _posts/en/newsletters/2024-02-14-newsletter.md:
```diff
@@ -0,0 +1,283 @@
+  for high-value channels.  Eclair's new default is the maximum feerate
+  suggested by its feerate estimator, up to an absolute total of 10,000
+  sat.  Eclair will also still pay up to the amount at risk from
+  [HTLCs][topic htlc] expiring in the near future, which could be higher
```
```suggestion
  [HTLCs][topic htlc] expiring soon, which could be higher
```

LarryRuane commented on _posts/en/newsletters/2024-02-14-newsletter.md:
```diff
@@ -0,0 +1,283 @@
+- [Bitcoin Core #28948][] adds support for [version 3 transaction
+  relay][topic v3 transaction relay] allowing any v3 transaction that
+  has no unconfirmed parent to enter the mempool according to the normal
+  transaction acceptance rules.  The v3 transaction can be [CPFP fee
+  bumped ][topic cpfp] but only if the child is 1,000 vbytes or less.
```
```suggestion
  transaction acceptance rules.  The v3 transaction can be [CPFP
  fee-bumped][topic cpfp] but only if the child is 1,000 vbytes or less.
```

LarryRuane commented on _posts/en/newsletters/2024-02-14-newsletter.md:
```diff
@@ -0,0 +1,283 @@
+  transactions defined in [BIP68][].  The new rules are intended to help
+  contract protocols such as LN ensure their precommitted transactions
+  can always be confirmed quickly with minimal extra fees needed to
+  escape [transaction pinning][topic transaction pinning].
```
```suggestion
  escape [transaction pinning][topic transaction pinning] attacks.
```

harding replied:
Leaving as-is.  There are legit cases where transaction pinning can happen by accident, especially if CPFP carve out wasn't a thing (which it might not be sooner or later), so pinning isn't always an attack.

LarryRuane commented on _posts/en/newsletters/2024-02-14-newsletter.md:
```diff
@@ -0,0 +1,295 @@
+[Bitcoin Inquisition][bitcoin inquisition repo], and [BINANAs][binana
+repo]._
+
+- [Bitcoin Core #28948][] adds support for (but does not enable) [version 3 transaction
+  relay][topic v3 transaction relay], allowing any v3 transaction that
```
Is it worth mentioning that the child (of a v3 transaction) does _not_ have to also be v3? Is that true? If the child _does_ have to also be v3, that might be worth mentioning.

Wait, what is the rule exactly? I couldn't tell from the PR description. Maybe it is:

> An unconfirmed v3 tx can have _either_ one unconfirmed parent _or_ one unconfirmed child (not both), and that parent or child can also be v3 but doesn't have to be. An unconfirmed v3 tx can also have no unconfirmed parents or children.

If that's correct, maybe consider stating that; it would have helped me understand it more easily.

glozow replied:
the child of a v3 tx *does* need to be v3.

harding commented:
Added lede, removed releases/RCs section (none this week), and added topic entries.  Reviewed review club section (thanks @LarryRuane!) and made all edits (thanks @glozow @glozow @LarryRuane !).  Thanks everyone!

bitschmidty reviewed (approved):
ACK topics, lede, and PR Review section updates

bitschmidty commented:
Squashed and merged, thank you @harding, @LarryRuane and team! 🚀

glozow commented on _posts/en/newsletters/2024-02-14-newsletter.md:
```diff
@@ -0,0 +1,295 @@
+
+  q6="Why can’t `maxfeerate` be checked immediately after decoding like
+      `maxburnamount` is?"
+  a6="Because transaction inputs famously don't explicitly state the
```
famously 🤣

LarryRuane replied:
Maybe I shouldn't have tried to be cute there, it just kind of slipped out, wasn't my fault! :grin:

ariard commented:
appreciated if there is a correction in future newsletters on Bitcoin Core #28948.
the merge has not been consensual according to Core review standards and the anti-pinning mitigation is weak
cf. my comment here: https://github.com/bitcoin/bitcoin/pull/28948#issuecomment-1946770281
